webrecorder/browsertrix
# Crawl config tag editor UI

Allow users to set tags on a crawl config. Resolves #362

### Manual testing
1. Run app with `yarn start`
2. Go to New Crawl Config page. Complete flow up to Crawl Information
   - Verify Tags field is shown
   - Verify you can add multiple tags and remove tags
   - Verify tags can be added to field by clicking "Add" button, hitting enter, or clicking away from the input area.
3. Save new crawl config. Verify tags are shown on the crawl config detail view.
4. Edit crawl config. Verify tags can be added and removed as expected.
5. Duplicate crawl config. Verify tags are duplicated

### Screenshots
**New Crawl Config - Crawl Information step:**
<img width="922" alt="Screen Shot 2023-01-10 at 8 23 07 PM" src="https://user-images.githubusercontent.com/4672952/211717607-9f85b61e-4946-41ef-9789-df2ca66f236d.png">

**On user input, hovered on "Add" button:**
<img width="165" alt="Screen Shot 2023-01-10 at 8 23 13 PM" src="https://user-images.githubusercontent.com/4672952/211717639-d6962026-d251-46a5-84c7-d9e45158ce0b.png">

**Entered tags:**
<img width="520" alt="Screen Shot 2023-01-10 at 8 23 19 PM" src="https://user-images.githubusercontent.com/4672952/211717669-39baf7de-447a-4e4f-8e5c-ca7a9b957fa9.png">

**On crawl config details (on crawls and crawl config view):**
<img width="904" alt="Screen Shot 2023-01-10 at 8 23 37 PM" src="https://user-images.githubusercontent.com/4672952/211717709-c2e93e29-c1a6-415b-9d5c-cb539027d890.png">

### Follow-ups
Autocomplete is ready to implement but still needs some back end changes. Tracked in separate issue here: https://github.com/webrecorder/browsertrix-cloud/issues/453
